caregiver_application/cdr/cdr_fhir.py

```diff
@@ -285,15 +286,24 @@ def get_patient(self, patient_id):
         except:
             return None
 
-    def get_observation(self, obsv_id):
-        try:
-            if not obsv_id.startswith("Observation/"):
-                obsv_id = "Observation/" + obsv_id
-            response = requests.get(f"{self.base_url}/{obsv_id}", headers=self.header)
-            return json.loads(response.text) if response else None
-        except:
-            return None
-
+    # def get_observation(self, obsv_id):
+    #     try:
+    #         if not obsv_id.startswith("Observation/"):
+    #             obsv_id = "Observation/" + obsv_id
+    #         response = requests.get(f"{self.base_url}/{obsv_id}", headers=self.header)
+    #         return json.loads(response.text) if response else None
+    #     except:
+    #         return None
+    
+    # get observations for a patient and a specific code is fetching in this method, 
+    def get_observations_for_patient_and_code(self, patient_id, observation_code):

```

Comment:
this function is too specific for a lib: I would prefer a function like find_observations (or query_observations) that has as second parameter the fhir search string. Should be similar as 'find_patient'. However, this name is also a bit disputable.

Reply:
The function is there to handle specific purpose, however, it is open to change if it confuses,